src/Protocol/ActivityPub/ClientToServer.php

```diff
@@ -0,0 +1,310 @@
+			Logger::debug('Empty body and title', ['id' => $id, 'uid' => $uid, 'object_id' => $object_id, 'activity' => $ldactivity]);
+			return [];
+		}
+		$post = ['title' => $item['title'], 'body' => $item['body']];
+		Logger::debug('Got data', ['id' => $id, 'uid' => $uid, 'item' => $post]);
+		Item::update($post, ['id' => $id]);
+		Item::updateDisplayCache($original_post['uri-id']);
+
+		return Transmitter::createActivityFromItem($id);
+	}
+
+	/**
+	 * Follow a given account
+	 * @todo Check the expected return value
+	 *
+	 * @param integer $uid
+	 * @param string $object_id
+	 * @param array $ldactivity
+	 * @return array
+	 */
+	private static function followAccount(int $uid, string $object_id, array $ldactivity): array
+	{
+		return [];
+	}
+
+	/**
+	 * Fetches data from the object part of an client to server activity
+	 *
+	 * @param array $object
+	 *
+	 * @return array Object data
+	 */
+	private static function processObject(array $object): array
+	{
+		$object_data = Receiver::getObjectDataFromActivity($object);
+
+		$object_data['target']   = self::getTargets($object, $object_data['actor'] ?? '');
+		$object_data['receiver'] = [];
+
+		return $object_data;
+	}
+
+	/**
+	 * Accumulate the targets and visibility of this post
+	 *
+	 * @param array $object
+	 * @param string $actor
+	 * @return array
+	 */
+	private static function getTargets(array $object, string $actor): array
+	{
+		$profile   = APContact::getByURL($actor);
+		$followers = $profile['followers'];
+
+		$targets = [];
+
+		foreach (['as:to', 'as:cc', 'as:bto', 'as:bcc'] as $element) {
+			switch ($element) {
+				case 'as:to':
+					$type = Receiver::TARGET_TO;
+					break;
+				case 'as:cc':
+					$type = Receiver::TARGET_CC;
+					break;
+				case 'as:bto':
+					$type = Receiver::TARGET_BTO;
+					break;
+				case 'as:bcc':
+					$type = Receiver::TARGET_BCC;
+					break;
+			}
+			$receiver_list = JsonLD::fetchElementArray($object, $element, '@id');
+			if (empty($receiver_list)) {
+				continue;
+			}
+
+			foreach ($receiver_list as $receiver) {
+				if ($receiver == Receiver::PUBLIC_COLLECTION) {
+					$targets[Receiver::TARGET_GLOBAL] = ($element == 'as:to');
+					continue;
+				}
+
+				if ($receiver == $followers) {
+					$targets[Receiver::TARGET_FOLLOWER] = true;
+					continue;
+				}
+				$targets[$type][] = Contact::getIdForURL($receiver);
+			}
+		}
+		return $targets;
+	}
+
+	/**
+	 * Create an item array from client to server object data
+	 *
+	 * @param array $object_data
+	 * @param array $application
+	 * @param integer $uid
+	 * @return array
+	 */
+	private static function processContent(array $object_data, array $application, int $uid): array
+	{
+		$owner = User::getOwnerDataById($uid);
+
+		$item = [];
+
+		$item['network']    = Protocol::DFRN;
+		$item['uid']        = $uid;
+		$item['verb']       = Activity::POST;
+		$item['contact-id'] = $owner['id'];
+		$item['author-id']  = $item['owner-id'] = Contact::getPublicIdByUserId($uid);
```

Comment:
Code standards from PHP CS Fixer 
```suggestion
		$item['author-id']  = $item['owner-id']  = Contact::getPublicIdByUserId($uid);
```